src/collection-browser.ts

```diff
@@ -1251,22 +1251,25 @@ export class CollectionBrowser
     return datePickerEnabled && (hasDateRange || hasYearFacet);
   }
 
-  private scrollToPage(pageNumber: number) {
-    const cellIndexToScrollTo = this.pageSize * (pageNumber - 1);
-    // without this setTimeout, Safari just pauses until the `fetchPage` is complete
-    // then scrolls to the cell
-    setTimeout(() => {
-      this.isScrollingToCell = true;
-      this.infiniteScroller.scrollToCell(cellIndexToScrollTo, true);
-      // This timeout is to give the scroll animation time to finish
-      // then updating the infinite scroller once we're done scrolling
-      // There's no scroll animation completion callback so we're
-      // giving it 0.5s to finish.
+  private scrollToPage(pageNumber: number): Promise<void> {
+    return new Promise(resolve => {
+      const cellIndexToScrollTo = this.pageSize * (pageNumber - 1);
+      // without this setTimeout, Safari just pauses until the `fetchPage` is complete
+      // then scrolls to the cell
       setTimeout(() => {
-        this.isScrollingToCell = false;
-        this.infiniteScroller.reload();
-      }, 500);
-    }, 0);
+        this.isScrollingToCell = true;
+        this.infiniteScroller?.scrollToCell(cellIndexToScrollTo, true);
+        // This timeout is to give the scroll animation time to finish
+        // then updating the infinite scroller once we're done scrolling
+        // There's no scroll animation completion callback so we're
+        // giving it 0.5s to finish.
```

Comment:
q: is animation via css or js? if js, do you think we should add callback on infinite scroller side?

Reply:
Hmm, I'm not actually sure -- I just wrapped this existing timeout stuff in a promise for ease of testing. Based on the comment, it sounds like the lack of a callback on infinite scroller may have been intentional, but I'll see if one can be cleanly added.